mozci/push.py

```diff
@@ -324,15 +336,16 @@ def tasks(self):
         # Skip tier tasks greater than the tier passed in config
         tasks = [task for task in tasks if not task.tier or task.tier <= config.tier]
 
-        # Now we can cache the results.
-        # cachy's put() overwrites the value in the cache; add() would only add if its empty
-        config.cache.put(
-            f"{self.push_uuid}/tasks",
-            tasks,
-            config["cache"]["retention"],
-        )
+        # Push is supposedly finalized, we can cache the results as tasks should've finish running
+        if self.is_finalized:
+            # cachy's put() overwrites the value in the cache; add() would only add if its empty
+            config.cache.put(
+                f"{self.push_uuid}/tasks",
+                tasks,
+                config["cache"]["retention"],
+            )
```

Comment:
A future optimization that we could do is to cache tasks, and then on the next run only used the cached tasks that were finished.
This way, we can avoid `Gathering test groups` for tasks that were already finished in the previous run and just do it for newly finished tasks.

Reply:
I filed #570 as a follow-up.